### PR TITLE
Add logic to fix absolute symlinks in / and notice about us doing this

### PR DESCRIPTION
**Pull request by troglodyne**
Took more time than I thought because I was busy getting frustrated trying to test till I gave up on it

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

File: lib/Elevate/Blockers/AbsoluteSymlinks.pm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+use Elevate::Components::AbsoluteSymlinks ();
+use Log::Log4perl qw(:easy);
+
+use parent qw{Elevate::Blockers::Base};
+
+sub check ($self) {
+    my %links = Elevate::Components::AbsoluteSymlinks::get_abs_symlinks();
+    WARN("Symlinks with absolute paths have been found in /:\n\t"
+        . join( ", ", keys(%links) )
```
Comment (atoomic) [inline lib/Elevate/Blockers/AbsoluteSymlinks.pm:24]: `sort`?

File: lib/Elevate/Components/AbsoluteSymlinks.pm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+sub pre_leapp {
+    my %links = get_abs_symlinks();
+    return unless %links;
+    my $chdir = Cpanel::Chdir->new("/");
+    foreach my $link (keys(%links)) {
+        my $updated = substr( $links{$link}, 1 );
+
+        my $rand_uid = Cpanel::UUID::random_uuid();
```
Comment (atoomic) [inline lib/Elevate/Components/AbsoluteSymlinks.pm:38]: note: even if the probability of collision is low, it can conflict...

Comment (toddr) [reply lib/Elevate/Components/AbsoluteSymlinks.pm:38]: Conflict with what? ideally nothing is happening on the system when they're upgrading?

Comment (troglodyne) [reply lib/Elevate/Components/AbsoluteSymlinks.pm:38]: yea... I realize nothing is ever truly "random" ever either... I'm just trying to *avoid* failures here when possible. Collision chances here should be so vanishingly small as to not ever be hit in the wild.

File: lib/Elevate/Blockers/AbsoluteSymlinks.pm (hunk 1 above)
Comment (troglodyne) [reply lib/Elevate/Blockers/AbsoluteSymlinks.pm:24]: Sure, makes sense

File: lib/Elevate/Components/AbsoluteSymlinks.pm (hunk 2 above)
Comment (toddr) [inline lib/Elevate/Components/AbsoluteSymlinks.pm:38]: check that the file is not there before you run symlink?

Comment (atoomic) [reply lib/Elevate/Components/AbsoluteSymlinks.pm:42]: that works

**Review by atoomic — APPROVED** [review]

**Pull request merged**